object-manipulation-3/main.js

```diff
@@ -0,0 +1,166 @@
+
+// -create a storage for all cards that have been "drawn".
+// -keep track of the number of cards to be dealt to each person.
+//   -keep track of the number of players.
+//     -check the random card to see if it's already been dealt. if it's one that's already been drawn,
+//       -then keep "shuffling" the card until a new one is generated.
+//     -add the random card into each player's hand.
+//     -add the random card into the list of dealt cards.
+//     -update the player's score with the score from the random card.
+
+// -create a function that will calculate a score based on a card's value.
+//   -note the first character in the card's name. if it's a number,
+//   then the score will be its corresponding number, except for number 10. if it's a face card or number 10,
+//   then the score will be 10, except for the Ace, which will have a score of 11.
+
+// -create a storage for the high score.
+// -look at each player.
+//   -compare the player's score with the current high score. if the player's score is higher, then replace the high score with the player's high score.
+
+// -create a storage for the winners.
+// -look at each player.
+//   -if the player has a score equivalent to the high score,
+//     -then add the player into the list of winners.
+
+// -declare the winners using the list of winners.
+
+// -create a function that will declare the winners.
+//   -if there is only one winner,
+//     -then write a message to declare the winner.
+//   -otherwise, if there are more,
+//     -create a storage for the names of the winners.
+//     -look at each of the winners.
+//       -if the current winner is not the last,
+//         -then add their name to the list of winners with the conjunction "and".
+//       -otherwise, if the current winner is the last one,
+//         -then add their name to the list of winners with an exclamation point.
+//     -write a message declaring the winners and the start of a new game.
+
+// -if the game ended in a tie and there are more than one winner,
+//   -then look at each of the winners.
+//    -clear each winner's hand.
+//    -clear each winner's score.
+//  -run the game again, dealing the same number of cards to the winners.
+
+playCards(generatePlayers(4), 2);
+
+function generatePlayers(number) {
+  var players = [];
+  var count = 1;
+
+  for (var i = 0; i < number; i++) {
+    players.push(
+      {
+        name: 'Player' + count,
+        hand: [],
+        score: null
+      }
+    );
+    count++;
+  }
+  return players;
+}
+
+function playCards(players, numberCards) {
+  var cardSuits = ['Hearts', 'Diamonds', 'Spades', 'Clubs'];
+  var cardRanks = ['Ace', 2, 3, 4, 5, 6, 7, 8, 9, 10, 'Jack', 'Queen', 'King'];
+  var randomCard = ((cardRanks[Math.floor(Math.random() * 13)]) + ' of ' + cardSuits[Math.floor(Math.random() * 4)]);
+
+  function shuffle() {
+    randomCard = ((cardRanks[Math.floor(Math.random() * 13)]) + ' of ' + cardSuits[Math.floor(Math.random() * 4)]);
+  }
```

Comment:
Rather than doing this, you should create a full deck of cards by looping over your array of suits and your array of ranks. You should make an object for each card with a `suit` and `rank` property so you can easily reference the information about the card, and code that references those values will be much more easily readable.